samloader/main.py

```diff
@@ -91,3 +91,8 @@ def getbinaryfile(client, fw, model, region):
     filename = root.find("./FUSBody/Put/BINARY_NAME/Data").text
     path = root.find("./FUSBody/Put/MODEL_PATH/Data").text
     return path, filename, size
+
+def getEncryptionVersion(file_name):
+    # Splits the filename and the extension // enc4 or enc2 is obtained
+    ext = file_name.split('.')[-1]
+    return 4 if ext == 'enc4' else 2
```

Comment:
How about making this code generic by simply returning the last character from `file_name` as the version?